### PR TITLE
Solve exercise

### LINKED ISSUE #15: [FEATURE] - Exercise View
### Reference Issues

#2 

### Summary

The view where students solve exercises. It should show all info related to an exercise, as well as a submit button that runs the code, a change language button, and "terminal" that shows compiler feedback.
image

### Unresolved questions

_No response_

### Definition of Done

When an exercise can be solved through the view.

### PR DESCRIPTION
**Pull request by andreaswurtz**
## Description

Anonymous users solve exercises.

### Resolved Issue
Fixes #15

### CODE REVIEW

File: src/routes/exercise/+page.server.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+async function postSolution(
+	backendUrl: string,
+	api_version: string,
+	access_token: string,
+	apiData,
+	exerciseId: int
+): Promise<Response> {
+	return await fetch(`${backendUrl}${api_version}/exercises/${exerciseId}/submission`, {
```
Comment (EMojoKrogh) [inline src/routes/exercise/+page.server.ts:83]: We need some refreshing on these fetches

File: src/routes/exercise/+page.server.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                }
+            } else {
+                resJSON = { detail: 'No response body' }; // Handle empty response body
+            }
+
+            console.log('resJSON:', resJSON); 
+            
+            if (resJSON.isFailed) { 
+                const errorMessages = resJSON.errors.map((err) => err.message).join('\n'); 
+                console.log('Epic fail from server:', resJSON);
```
Comment (EMojoKrogh) [inline src/routes/exercise/+page.server.ts:146]: xD

**Review by EMojoKrogh — APPROVED** [review]
Looks like svelte code

**Review by EMojoKrogh — CHANGES_REQUESTED** [review]
In fetches change "fetch(`${backendUrl}${apiVersion}" to fetch(`${backendUrl}/${apiVersion}

File: src/routes/exercise/+page.server.ts (hunk 1 above)
Comment (andreaswurtz) [reply src/routes/exercise/+page.server.ts:83]: Would require a new method as the request handler doesn't seem to work for anon tokens?

**Comment by andreaswurtz** [comment]
For some reason i'm getting this error after merging main into the branch. 
```
TypeError: Cannot read properties of undefined (reading 'name')
    at load (/home/andreas/Programming/P7/frontend/src/routes/+layout.server.ts:27:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Module.load_server_data (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:61:17)
    at async eval (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:156:13)
TypeError: Cannot read properties of undefined (reading 'name')
    at load (/home/andreas/Programming/P7/frontend/src/routes/+layout.server.ts:27:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Module.load_server_data (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:61:17)
    at async Module.respond_with_error (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/page/respond_with_error.js:57:24)
    at async resolve (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/respond.js:524:12)
    at async Module.respond (/home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/runtime/server/respond.js:327:20)
    at async file:///home/andreas/Programming/P7/frontend/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:524:22
```

**Review by EMojoKrogh — APPROVED** [review]
Gangster

**Pull request merged**